src/main/java/reposense/report/RepoCloner.java

```diff
@@ -0,0 +1,162 @@
+import reposense.system.LogsManager;
+import reposense.util.FileUtil;
+
+public class RepoCloner {
+    private static final int MAX_NO_OF_REPOS = 2;
+    private static final Logger logger = LogsManager.getLogger(RepoCloner.class);
+
+    private RepoConfiguration[] configs = new RepoConfiguration[MAX_NO_OF_REPOS];
+    private int index = 0;
+    private int prevIndex = 0;
+    private boolean isCurrentRepoCloned = false;
+    private String prevRepoDefaultBranch;
+    private CommandRunnerProcess crp;
+
+    /**
+     * Spawns a process to clone the repository specified by {@code config}.
+     * Does not wait for process to finish executing.
+     */
+    public void clone(String outputPath, RepoConfiguration config) throws IOException {
+        configs[index] = config;
+        isCurrentRepoCloned = true;
+        if (isPreviousRepoDifferent()) {
+            isCurrentRepoCloned = spawnCloneProcess(outputPath, config);
+        }
+    }
+
+    /**
+     * Waits for current clone process to finish executing and returns the corresponding {@code RepoConfiguration}.
+     */
+    public RepoConfiguration getClonedRepo(String outputPath) throws IOException {
+        if (isCurrentRepoCloned && isPreviousRepoDifferent()) {
+            isCurrentRepoCloned = waitForCloneProcess(outputPath, configs[index]);
+        }
+
+        if (!isCurrentRepoCloned) {
+            deleteDirectory(configs[index].getRepoRoot());
+            return null;
+        }
+
+        if (isPreviousRepoDifferent()) {
+            prevRepoDefaultBranch = GitBranch.getCurrentBranch(configs[index].getRepoRoot());
+        } else {
+            configs[index].updateBranch(prevRepoDefaultBranch);
+        }
+        cleanupPrevRepoFolder();
+
+        try {
+            GitCheckout.checkout(configs[index].getRepoRoot(), configs[index].getBranch());
+        } catch (RuntimeException e) {
+            logger.log(Level.SEVERE, "Branch does not exist! Analysis terminated.", e);
+            generateEmptyRepoReport(outputPath, configs[index]);
+            return null;
+        }
+        prevIndex = index;
+        index = (index + 1) % configs.length;
+        return configs[prevIndex];
+    }
+
+    /**
+     * Cleans up after all repos have been cloned and analyzed
+     */
+    public void cleanup() {
+        deleteDirectory(FileUtil.REPOS_ADDRESS);
+    }
+
+    /**
+     * Returns true if current repo is different from the previously cloned repo.
+     */
+    private boolean isPreviousRepoDifferent() {
+        return prevIndex == index || !configs[prevIndex].getLocation().equals(configs[index].getLocation());
+    }
+
+    /**
+     * Spawns a process to clone repo specified in {@code repoConfig}. Does not wait for process to finish executing.
+     * Should only handle a maximum of one spawned process at any time.
+     */
+    private boolean spawnCloneProcess(String outputPath, RepoConfiguration config) throws IOException {
```

Comment:
The reason I requested him to make it a member method because he need to keep track of the `crp` in order to check its status. Moving it to elsewhere would delegate that duty / add unnecessary overhead, let's just keep it within RepoCloner since it is its main responsibility.